semantic-core/src/Analysis/Eval.hs

```diff
@@ -33,7 +33,7 @@ eval :: ( Carrier sig m
         , MonadFail m
         , Semigroup value
         )
-     => Analysis address value m
+     => Analysis (Term (Ann :+: Core) User) address value m
```

Comment:
`eval` is still specialized to `Term (Ann :+: Core) User`; intuitively, it is the canonical abstract interpreter for `Core`, and thus _has_ to be specialized to it.